base/applications/osk/settings.c

```diff
@@ -90,6 +97,34 @@ BOOL LoadDataFromRegistry()
 
     /* Load the sound on click value event */
     Globals.bSoundClick = (dwSoundOnClick != 0);
+
+    /* Query the key */
+    lResult = RegQueryValueExW(hKey,
+                               L"DialogPlacement",
+                               NULL,
+                               &dwType,
+                               (LPBYTE)Buff,
+                               &dwSize);
+
+    if (lResult != ERROR_SUCCESS || dwType != REG_SZ)
+    {
+        /* Bail out and return FALSE if we fail */
+        RegCloseKey(hKey);
+        return FALSE;
+    }
+
+    /* 
+        Parse the data into the buffer and check if such data is valid.
+        Otherwise the dialog will be placed at the bottom of the screen.
+     */
+    if (swscanf(Buff, L"%d %d",
+        &wp.rcNormalPosition.left,
+        &wp.rcNormalPosition.top) == 2)
```

Comment:
Why is `wp` used as scratch here?

Reply:
What's wrong with `wp` in the code? What do you mean by `scratch`?

Reply:
I think he is asking you why you used a full wp variable (as a temporary variable) for just retrieving two LONG left and top variables.